ttnn/cpp/ttnn/operations/matmul/matmul.cpp

```diff
@@ -92,6 +92,79 @@ ttnn::Tensor bound_matmul(
     return output_tensor;
 }
 
+Tensor MatmulOperation::invoke(
+    const Tensor& input_tensor_a,
+    const Tensor& input_tensor_b,
+    const bool transpose_a,
+    const bool transpose_b,
+    const MemoryConfig& memory_config,
+    const std::optional<const DataType> dtype,
+    const std::optional<const MatmulProgramConfig> program_config,
+    const std::optional<const std::string>& activation,
+    const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
+    const std::optional<const CoreGrid> core_grid) {
+    std::optional<CoreCoord> user_core_coord;
+    if (core_grid.has_value()) {
+        user_core_coord = CoreCoord(core_grid->x, core_grid->y);
+    }
+    bool user_run_batched = detail::is_input_batched(input_tensor_b.get_shape());
+    return bound_matmul(
```

Comment:
Does this bound_matmul calls launch_op inside or only operation::run?

Reply:
luanch_op is used.

bound_matmul does a bunch of stuff:
- it may transpose the vectors
- it calls the underlying matmul, which calls 
```
...
operation::launch_op(
...
  return operation::run(...
```
- it may call ttnn::add()
- it may call activation methods (ttnn::relu/gelu/silu)